src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -251,57 +262,95 @@ private FormData prepareFormDataSet(FormFields form) {
 
 	/**
 	 * Gets the values for the given {@code field}.
-	 * <p>
-	 * If the field is of submit type it returns its predefined values.
+	 * If the field is of submit type it passes the predefined values to the DefaultValueGenerator and returns its predefined values.
+	 * Gets the default value that the input field, including HTML5 types, should have.
 	 *
 	 * @param field the field
 	 * @return a list with the values
-	 * @see #getDefaultTextValue(FormField)
 	 */
-	private List<String> getValues(FormField field) {
-		if (field.getFormControl().getFormControlType().isSubmit()) {
-			return new ArrayList<>(field.getPredefinedValues());
-		}
+	private List<String> getDefaultTextValue(FormField field) {
 
 		// Get its value(s)
 		List<String> values = field.getValues();
```

Comment:
Could now be moved to after the if "Handles Submit Fields", it's only needed after that if.